indico/htdocs/js/indico/jquery/palettepicker.js

```diff
@@ -0,0 +1,116 @@
+ */
+
+(function($) {
+    $.widget('indico.palettepicker', {
+
+        options: {
+            availableColors: [],
+            onSelect: null,
+            numColumns: 5
+        },
+
+        _create: function() {
+            var self = this;
+            var element = this.element;
+            var palette = $('<div>', {'class': 'color-palette'});
+            var paletteTable = $('<table>', {'class': 'palette-picker'});
+            var availableColors = this.options.availableColors;
+            var tr = this._createTableRow();
+
+            for (var i = 0; i < availableColors.length; ++i) {
+                var uniqueId = _.uniqueId('palette-color');
+                var td = $('<td>', {
+                    'css': {'background': '#' + availableColors[i].background},
+                    'class': 'palette-color',
+                    'id': uniqueId
+                });
+
+                $(document).on('click', '#' + uniqueId, function(evt) {
+                    evt.preventDefault();
+
+                    var $this = $(this),
+                        backgroundColor = $this.find('.background-box').data('value'),
+                        textColor = $this.find('.text-box').data('value');
+
+                    element.css({'background': backgroundColor, 'color': textColor + ' !important'});
+
+                    if (self.options.onSelect) {
+                        self.options.onSelect.call(element, backgroundColor, textColor);
+                    }
+
+                    element.qtip('hide');
+                });
+
+                var colorBox = $('<div>', {
+                    'data-value': '#' + availableColors[i].background,
+                    'class': 'background-box'
+                });
+
+                colorBox.append($('<div>', {
+                    'css': {'background': '#' + availableColors[i].text},
+                    'class': 'text-box',
+                    'data-value': '#' + availableColors[i].text
+                }));
+
+                td.append(colorBox);
+                tr.append(td);
+
+                if ((i + 1) % this.options.numColumns == 0) {
+                    paletteTable.append(tr);
+                    tr = this._createTableRow();
+                }
+            }
+
+            if (tr.children().length) {
+                paletteTable.append(tr);
+            }
+
+            palette.append(paletteTable);
+
+            element.qtip({
+                prerender: false,
+                overwrite: false,
+                style: {
+                    classes: 'color-palette'
+                },
+                position: {
+                    my: 'top center',
+                    at: 'bottom center',
+                    target: element,
+                    adjust: {
+                        mouse: false,
+                        scroll: false
+                    }
+                },
+                content: {
+                    text: palette.html()
```

Comment:
I think you can pass a jquery object here directly (and not lose e.g. event handlers that way)